pkgs/top-level/release-cuda.nix

```diff
@@ -68,12 +71,61 @@ let
     pkgs
     ;
 
+  # Packages that match the following predicates won't be evaluated on any platforms
+  # Each predicate takes a human-readable, dot-delimited attribute path and the derivation itself
+  filterPackagePredicates =
+    let
+      matchCudaPackageByName = name: path: null != builtins.match "cudaPackages[^\.]*\.${name}" path;
+      knownBrokenConditions =
+        conds: drv: builtins.any (cond: drv.brokenConditions.${cond} or false) conds;
+    in
+    [
+      # TensorRT currently can't be built on hydra/nix-community builders due to requiring the user
+      # to explicitly agree to the NVIDIA TensorRT License (and a developer account on developer.nvidia.com)
+      # See https://docs.nvidia.com/deeplearning/tensorrt/latest/reference/sla.html
+      (drv: matchCudaPackageByName "tensorrt[^\.]*")
```

Comment:
I wonder if we could limit ourselves to `lib.strings.hasPrefix`

Reply:
I don't think `hasPrefix` works too well here, because we want to match not only `cudaPackages.tensorrt`, but also `cudaPackages_11.tensorrt`, `cudaPackages.tensorrt_10_3`, `cudaPackages_11.tensorrt_10_3`, etc.

`mapAttrsRecursive` provides each attribute path as a list (like `["cudaPackages_11", "tensorrt_10_3"]`), so we could hypothetically match/unpack attribute paths of lenght 2 and then check the first attrname with `hasPrefix "cudaPackages"` and second attrname with `hasPrefix "tensorrt"`, but I am not sure if that is simpler than just a single regex match.

Reply:
Ah right this acts on attribute paths. Well the reason I'm hesitant to use regex is that there's recently been a conversation on matrix about libc++ and libstdc++ (and so builtins.match) regexps having different behaviour in some edge cases, and I thought maybe it's a good idea to just start avoiding them generally

Reply:
Since I removed `filterPackagePredicates`, this is no longer relevant. Marking as resolved for now.